x/leverage/keeper/keeper.go

```diff
@@ -108,33 +108,15 @@ func (k Keeper) LendAsset(ctx sdk.Context, lenderAddr sdk.AccAddress, loan sdk.C
 }
 
 // WithdrawAsset attempts to deposit uTokens into the leverage module in exchange
-// for the original tokens loaned. Accepts either a uToken amount to withdraw or
-// an equivalent base token amount to be converted automatically via exchange rate.
-// If the token or uToken denom is invalid or account balance insufficient for either
-// lender or module, we return an error.
-func (k Keeper) WithdrawAsset(ctx sdk.Context, lenderAddr sdk.AccAddress, withdrawal sdk.Coin) error {
-	var (
-		uToken sdk.Coin
-		err    error
-	)
-
-	if k.IsAcceptedToken(ctx, withdrawal.Denom) {
-		// Automatically convert base token input to equivalent uTokens
-		uToken, err = k.ExchangeToken(ctx, withdrawal)
```

Comment:
seams like we had a potential bug here if we the function doesn't handle case where `uToken == withdrawal`.